contracts/mainnet/connectors/uniswapV3/main.sol

```diff
@@ -0,0 +1,184 @@
+pragma solidity ^0.7.6;
+pragma abicoder v2;
+
+/**
+ * @title Uniswap v3.
+ * @dev Decentralized Exchange.
+ */
+
+import {TokenInterface} from "../../common/interfaces.sol";
+import {Helpers} from "./helpers.sol";
+import {Events} from "./events.sol";
+
+abstract contract UniswapResolver is Helpers, Events {
+    /**
+     * @dev Mint New Position
+     * @notice Mint New NFT LP Position
+     * @param params parameter for mint
+     * @param getIds ID to retrieve amtA
+     * @param setId ID stores the amount of LP token
+     */
+    function mint(
+        MintParams memory params,
```

Comment:
Hey @cryptoDev222! Why struct here and nowhere else? Would be good if we convert the struct here into normal variables too.

Is it the problem with too many variables? Then maybe we can create one more internal function which will take struct and this function will mere all these variables into a struct and pass.

Reply:
Yess, Since we should pass too many variables.
Even if we made one internal function, we should pass these variables once, then it causes stack too deep error.

Reply:
Okay. Maybe leave it like that.